### PR TITLE
JCSP-389: DC, Cloud: Quick search top bar

### PR DESCRIPTION
**Pull request by rsaliiatlassian**

### CODE REVIEW

File: custom-vu/src/main/kotlin/jces1209/vu/ScenarioSimilarities.kt [hunk 1]
```diff
@@ -89,6 +91,11 @@ class ScenarioSimilarities(
             meter = meter,
             scrumBoardsMemory = boardsMemory.scrum,
             jiraTips = JiraTips(jira.driver)
+        ),
+        workOnTopBar = WorkOnTopBar(
```
Comment (dagguh) [inline custom-vu/src/main/kotlin/jces1209/vu/ScenarioSimilarities.kt:105]: Adding it on the end increases risk of Git conflict -> unnecessary work.
Worth keeping in mind for the next time.

File: custom-vu/src/main/kotlin/jces1209/vu/page/bars/topBar/AbstractTopBar.kt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package jces1209.vu.page.bars.topBar
+
+interface AbstractTopBar {
+    fun quickSearch(): AbstractTopBar
+}
```
Comment (dagguh) [inline custom-vu/src/main/kotlin/jces1209/vu/page/bars/topBar/AbstractTopBar.kt]:
```suggestion
interface TopBar {
    fun quickSearch(): TopBar
}
```

**Review by dagguh — DISMISSED** [review]

File: custom-vu/src/main/kotlin/jces1209/vu/action/WorkOnTopBar.kt [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import com.atlassian.performance.tools.jiraactions.api.measure.ActionMeter
+import jces1209.vu.page.bars.topBar.AbstractTopBar
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+
+/**
+ * Works for both Cloud and Data Center.
+ */
+class WorkOnTopBar(
+    private val topBar: AbstractTopBar,
```
Comment (ysemenkov) [inline custom-vu/src/main/kotlin/jces1209/vu/action/WorkOnTopBar.kt]: It must be compilation error - there is no AbstractTopBar - it's just `TopBar` already

File: custom-vu/src/main/kotlin/jces1209/vu/action/WorkOnTopBar.kt [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+) : Action {
+    private val logger: Logger = LogManager.getLogger(this::class.java)
+
+    override fun run() {
+        jira.navigateTo("secure/BrowseProjects.jspa")
+        quickSearch(topBar)
+    }
+
+    private fun quickSearch(topBar: AbstractTopBar) {
+        meter.measure(ActionType("Quick search top bar") { Unit }) {
```
Comment (ysemenkov) [inline custom-vu/src/main/kotlin/jces1209/vu/action/WorkOnTopBar.kt]: Please create it in MeasureType

File: custom-vu/src/main/kotlin/jces1209/vu/page/bars/topBar/cloud/CloudTopBar.kt [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+                ExpectedConditions.presenceOfAllElementsLocatedBy(By.xpath("//*[@data-test-id='search-dialog-dialog-wrapper']//*[. = 'Recently viewed issues']"))
+            )
+        driver
+            .wait(
+                ExpectedConditions.presenceOfAllElementsLocatedBy(By.xpath("//*[@data-test-id='search-dialog-dialog-wrapper']//span[@title]"))
+            )
+        driver
+            .wait(
+                ExpectedConditions.presenceOfAllElementsLocatedBy(By.xpath("//*[@data-test-id='search-dialog-dialog-wrapper']//*[. = 'Boards, Projects and Filters']"))
+            )
```
Comment (ysemenkov) [inline custom-vu/src/main/kotlin/jces1209/vu/page/bars/topBar/cloud/CloudTopBar.kt]: ExpectedConditions.and

File: custom-vu/src/main/kotlin/jces1209/vu/action/WorkOnTopBar.kt [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+ */
+class WorkOnTopBar(
+    private val topBar: AbstractTopBar,
+    private val jira: WebJira,
+    private val meter: ActionMeter
+) : Action {
+    private val logger: Logger = LogManager.getLogger(this::class.java)
+
+    override fun run() {
+        jira.navigateTo("secure/BrowseProjects.jspa")
```
Comment (ysemenkov) [inline custom-vu/src/main/kotlin/jces1209/vu/action/WorkOnTopBar.kt:22]: also, it's necessary to add `wait for page load`

File: custom-vu/src/main/kotlin/jces1209/vu/page/bars/topBar/dc/DcTopBar.kt [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+            )
+        driver
+            .wait(
+                ExpectedConditions.presenceOfAllElementsLocatedBy(By.xpath("//*[@aria-tooltip='View all issues']"))
+            )
+        driver
+            .wait(
+                ExpectedConditions.presenceOfAllElementsLocatedBy(By.xpath("//*[@aria-tooltip='View all projects']"))
+            )
+        return this
```
Comment (ysemenkov) [inline custom-vu/src/main/kotlin/jces1209/vu/page/bars/topBar/dc/DcTopBar.kt]: ExpectedConditions.and

**Review by ysemenkov — CHANGES_REQUESTED** [review]

**Review by ysemenkov — DISMISSED** [review]

**Review by ysemenkov — APPROVED** [review]

**Pull request merged**